compiler/modules/front/codeGen.bal

```diff
@@ -389,13 +389,13 @@ function codeGenForeachStmt(CodeGenContext cx, bir:BasicBlock startBlock, Enviro
     initLoopVar.insns.push(init);
     bir:BasicBlock loopHead = cx.createBasicBlock();
     bir:BasicBlock exit = cx.createBasicBlock();
-    bir:BranchInsn branchToLoopHead = { dest: loopHead.label };
+    bir:BranchInsn branchToLoopHead = { dest: loopHead.label, pos: stmt.startPos };
```

Comment:
Using the start position of the stmt feels a bit weird to me in these insns. Not sure what the right answer is.

Reply:
changed the `pos` of `BranchInsn` to the start of the loop to `stmt.endPos` (because we use it at the end of the loop) and kept the `pos` of `CondBranchInsn` at `stmt.startPos` (because we use it at the beginning of the loop)